### PR TITLE
Fix compiler messages

### PR DESCRIPTION
**Pull request by IhorNehrutsa**

### CODE REVIEW

**Comment by FedericoBusero** [comment]
Can you give more information on used board, version of Arduino core, ... to know which compile errors you want to solve?

**Comment by IhorNehrutsa** [comment]
Simply click "Platformio:Build" and select "PROBLEMS" tab.
image

File: lib/audio_output/src/DACOutput.cpp [hunk 1]
```diff
@@ -2,15 +2,15 @@
...
 {
     // i2s config for writing both channels of I2S
     i2s_config_t i2s_config = {
         .mode = (i2s_mode_t)(I2S_MODE_MASTER | I2S_MODE_TX | I2S_MODE_DAC_BUILT_IN),
         .sample_rate = sample_rate,
         .bits_per_sample = I2S_BITS_PER_SAMPLE_16BIT,
         .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT,
-        .communication_format = (i2s_comm_format_t)(I2S_COMM_FORMAT_I2S_MSB),
+        .communication_format = I2S_COMM_FORMAT_STAND_I2S,
         .intr_alloc_flags = ESP_INTR_FLAG_LEVEL1,
```
Comment (IhorNehrutsa) [inline lib/audio_output/src/DACOutput.cpp:14]: I2S_COMM_FORMAT_I2S is deprecated, correspond to `I2S_COMM_FORMAT_STAND_I2S`

File: lib/audio_output/src/DACOutput.cpp [hunk 2]
```diff
@@ -2,15 +2,15 @@
...
+void DACOutput::start(unsigned int sample_rate)
 {
     // i2s config for writing both channels of I2S
     i2s_config_t i2s_config = {
         .mode = (i2s_mode_t)(I2S_MODE_MASTER | I2S_MODE_TX | I2S_MODE_DAC_BUILT_IN),
         .sample_rate = sample_rate,
         .bits_per_sample = I2S_BITS_PER_SAMPLE_16BIT,
         .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT,
-        .communication_format = (i2s_comm_format_t)(I2S_COMM_FORMAT_I2S_MSB),
+        .communication_format = I2S_COMM_FORMAT_STAND_I2S,
```
Comment (IhorNehrutsa) [inline lib/audio_output/src/DACOutput.cpp:13]: I2S_COMM_FORMAT_I2S is deprecated, correspond to I2S_COMM_FORMAT_STAND_I2S

File: src/config.h [hunk 3]
```diff
@@ -47,17 +47,20 @@
...
 #define ESP_NOW_WIFI_CHANNEL 1
 
-// In case all transport packets need a header (to avoid interference with other applications or walkie talkie sets), 
+// In case all transport packets need a header (to avoid interference with other applications or walkie talkie sets),
 // specify TRANSPORT_HEADER_SIZE (the length in bytes of the header) in the next line, and define the transport header in config.cpp
 #define TRANSPORT_HEADER_SIZE 0
 extern uint8_t transport_header[TRANSPORT_HEADER_SIZE];
 
 
+#ifndef USE_I2S_MIC_INPUT
```
Comment (IhorNehrutsa) [inline src/config.h:56]: This is just for highlighting in VS Code.
This makes the code easier to read.

File: src/config.cpp [hunk 4]
```diff
@@ -5,7 +5,7 @@
 // For example, when TRANSPORT_HEADER_SIZE is defined as 3,  define transport_header for example as {0x1F, 0xCD, 0x01};
 uint8_t transport_header[TRANSPORT_HEADER_SIZE] = {};
 
-
+#ifndef USE_I2S_MIC_INPUT
```
Comment (IhorNehrutsa) [inline src/config.cpp:8]: This is just for highlighting in VS Code.
This makes the code easier to read.

**Comment by IhorNehrutsa** [comment]
I don't like  #ifndef directive, it is easy to confuse with #ifdef

Але #ifdef потребує переміщення рядків в коді, і я вирішив не рухати рядки, а використати #ifdef

**Comment by IhorNehrutsa** [comment]
Yes, I have ESP32-S3 and the next fix will be dedicated to the ESP32-S3

File: lib/audio_output/src/DACOutput.cpp (hunk 2 above)
Comment (FedericoBusero) [reply lib/audio_output/src/DACOutput.cpp:13]: I'll update the code to the IDF 4.4.x i2s API (Arduino ESP32 core 2.0.8) in the next days, but I still want to be compatible with ESP32 core 1.0.6 too, so I will add some ifdef´s. Also I´d like to test on S2/S3 and fix the compilation issues.

**Comment by FedericoBusero** [comment]
Thanks for your PR. I took your ideas, but implemented the changes backward compatible with older IDF versions.
- DACOutput : I2S_COMM_FORMAT_I2S_MSB -> I2S_COMM_FORMAT_STAND_MSB instead of I2S_COMM_FORMAT_STAND_I2S
Although your suggestion compiles, there was no audio on DAC output, so I changed it to the correct value

- DACOutput, I2SOutput, Output : int sample_rate -> uint32_t instead of unsigned int
For most chips uint32_T and unsigned int are the same, but this is not guaranteed (e.g. RISC-V ESP32-C3)

- config.cpp  : mic_config: I2S_COMM_FORMAT_I2S ->I2S_COMM_FORMAT_STAND_I2S

I will close it now, in case there are other issues, please tell me. I checked the code on ESP32-C3, S2, ESP32 but not yet on S3, but the code is according to the documentation.

**Pull request closed without merging**